ocaml/typing/ctype.ml

```diff
@@ -2143,6 +2205,23 @@ let rec estimate_type_jkind env ty =
   | Tpoly (ty, _) -> estimate_type_jkind env ty
   | Tpackage _ -> Jkind (Type.Primitive.value ~why:First_class_module |> Jkind.of_type_jkind)
 
+and type_jkind env ty =
+  jkind_of_result (estimate_type_jkind env (get_unboxed_type_approximation env ty))
+
+let arity_matches_decl env decl t = match t, decl.type_arity with
```

Comment:
I think this now only has one valid call site for this function: the one in `typetexp.ml`. It should possibly be moved in there, renamed `check_arity_matches_decl` and changed to raise the associated error itself directly.